src/api/procedures/createPortfolios.ts

```diff
@@ -2,71 +2,80 @@ import { ISubmittableResult } from '@polkadot/types/types';
 
 import { Context, NumberedPortfolio, PolymeshError, Procedure } from '~/internal';
 import { ErrorCode, TxTags } from '~/types';
-import { BatchTransactionSpec, ExtrinsicParams } from '~/types/internal';
+import { BatchTransactionSpec, ProcedureAuthorization } from '~/types/internal';
 import {
   identityIdToString,
   stringToBytes,
   stringToIdentityId,
   u64ToBigNumber,
 } from '~/utils/conversion';
-import { filterEventRecords, getPortfolioIdsByName } from '~/utils/internal';
-
+import { checkTxType, filterEventRecords, getPortfolioIdsByName } from '~/utils/internal';
+type PortfolioParams = {
+  name: string;
+  ownerDid?: string;
+};
 /**
  * @hidden
  */
 export interface Params {
-  names: string[];
+  portfolios: PortfolioParams[];
 }
-
 /**
  * @hidden
  */
 export const createPortfoliosResolver =
   (context: Context) =>
   (receipt: ISubmittableResult): NumberedPortfolio[] => {
     const records = filterEventRecords(receipt, 'portfolio', 'PortfolioCreated');
-
     return records.map(({ data }) => {
       const did = identityIdToString(data[0]);
       const id = u64ToBigNumber(data[1]);
-
       return new NumberedPortfolio({ did, id }, context);
     });
   };
-
 /**
  * @hidden
  */
 export async function prepareCreatePortfolios(
   this: Procedure<Params, NumberedPortfolio[]>,
   args: Params
-): Promise<
-  BatchTransactionSpec<NumberedPortfolio[], ExtrinsicParams<'portfolio', 'createPortfolio'>[]>
-> {
+): Promise<BatchTransactionSpec<NumberedPortfolio[], unknown[][]>> {
```

Comment:
I think this can get a better type than `unknown`. Should be the union of the two possible transactions